nunaliit2-js/src/main/js/nunaliit2/n2.widgetLegend.js

```diff
@@ -28,558 +28,1306 @@ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+			if( currentClasses ) {
+				classNames = currentClasses.split(' ');
+			};
+
+			if( classNames.indexOf(className) < 0 ){
+				classNames.push(className);
+			};
+
+			elem.setAttribute('class',classNames.join(' '));
+		},
+
+		/**
+		 * Method: _computePathFromSymbol Given an OpenLayers symbol (array of
+		 * points, which are tuples of x,y coordinates), create a SVG path with an
+		 * approximate area of 30 (area of a circle with a radius of 5) Example for
+		 * symbol: [0,0, 1,0, 1,1, 0,1, 0,0] Examplke fo SVG Path: 'M -4.4 -4.4 L
+		 * -4.4 4.4 L 4.4 4.4 L 4.4 -4.4 Z'
+		 */
+		_computePathFromSymbol: function(symbol){
+			var area = 0,
+			minx = undefined,
+			maxx = undefined,
+			miny = undefined,
+			maxy = undefined;
+
+			// Figure out bounding box
+			for(var i=0,e=symbol.length; i<e; i=i+2){
+				var x = symbol[i];
+				var y = symbol[i+1];
+
+				if( typeof minx === 'undefined' ){
+					minx = x;
+				} else if( minx > x ){
+					minx = x;
+				};
+
+				if( typeof maxx === 'undefined' ){
+					maxx = x;
+				} else if( maxx < x ){
+					maxx = x;
+				};
+
+				if( typeof miny === 'undefined' ){
+					miny = y;
+				} else if( miny > y ){
+					miny = y;
+				};
+
+				if( typeof maxy === 'undefined' ){
+					maxy = y;
+				} else if( maxy < y ){
+					maxy = y;
+				};
+			};
+
+			// Compute path, recentering the symbol and adjusting the area so
+			// it fits a bounding box of 10x10
+			var path = [],
+			transx = (minx+maxx)/2,
+			transy = (miny+maxy)/2,
+			width = maxx-minx,
+			height = maxy-miny,
+			factor = (width > height) ? width / 10 : height / 10;
+			if( factor <= 0 ){
+				factor = 1;
+			};
+			for(var i=0,e=symbol.length; i<e; i=i+2){
+				var x = symbol[i];
+				var y = symbol[i+1];
+
+				var effX = (x-transx)/factor;
+				var effY = (y-transy)/factor;
+
+				// Round to .01
+				effX = Math.floor(effX * 100) / 100;
+				effY = Math.floor(effY * 100) / 100;
+
+				if( 0 === i ){
+					path.push('M ');
+				} else {
+					path.push('L ');
+				};
+
+				path.push(''+effX);
+				path.push(' '+effY+' ');
+			};
+			path.push('Z');
+
+			return path.join('');
+		}
+	});
+
+	// LegendWidget2 The widget draws a legendWidget based on the styleRules defined inside map.json
+	// Also, this widget has a checkbox before each entry, which allows users to filter in/out the documents based on the style condition.
+	// This widget connects with a conditionalModelFilter to do the model filtering. 
+	// Two model parameters: completeChoiceParameter and selectedChoiceParameter is the messenger
+	// between this widget and the conditionalModelFilter mentioned above
+	// @Param containerId {string}
+	// @Param sourceModelId {string} sourceModelId of the conditionalModelFilter
+	// @Param labels {array} Optional parameter: if provided, only the label inside this list (must also defined valid condition inside map.json)
+	// will be rendered inside widgetLegend.
+	
+	var LegendWidget2 = $n2.Class('LegendWidget2', {
```

Comment:
legendWidget2 doesn't seem to work with custom conditions (e.g. iran atlas's use of isMajorityLanguage condition function). When you uncheck one of these items in the legend widget, no-changes occur in the map.

Not sure if widget code needs to be updated, or if these custom style rules need to be updated to be used with the new widget.

Reply:
Custom conditions is supported. In order for it to work with custom condition, we need to create two pair of each condition function: One of them executes upon geometry, the other executes upon doc. And install them on styleGlobalContext individually. This should be inside nunaliit_custom.js. The same place where custom condition functions are defined